src/components/step-flow/step-flow.pw.tsx

```diff
@@ -0,0 +1,275 @@
+      mount,
+      page,
+    }) => {
+      await mount(
+        <StepFlowComponent totalSteps={totalSteps} showProgressIndicator />
+      );
+
+      expect(await stepFlowProgressIndicator(page).count()).toEqual(totalSteps);
+    });
+  });
+
+  test("should render the correct element when 'showCloseIcon' is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<StepFlowComponent showCloseIcon />);
+
+    await expect(stepFlowDismissIcon(page)).toBeVisible();
+  });
+
+  test("background colour token should be correct when 'showCloseIcon' is true and the indicator is in progress", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <StepFlowComponent currentStep={1} totalSteps={1} showProgressIndicator />
+    );
+
+    const backgroundColorToken = await getDesignTokensByCssProperty(
+      page,
+      stepFlowProgressIndicator(page),
+      "background-color"
+    );
+
+    expect(backgroundColorToken[0]).toBe("--colorsUtilityYin090");
+  });
+
+  test("background colour token should be correct when 'showCloseIcon' is true and the indicator is completed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <StepFlowComponent currentStep={2} totalSteps={3} showProgressIndicator />
+    );
+
+    const backgroundColorToken = await getDesignTokensByCssProperty(
+      page,
+      stepFlowProgressIndicator(page).nth(0),
+      "background-color"
+    );
+
+    expect(backgroundColorToken[0]).toBe("--colorsSemanticPositive500");
+  });
+
+  test("background colour should be correct when 'showCloseIcon' is true and the indicator is not completed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <StepFlowComponent currentStep={2} totalSteps={3} showProgressIndicator />
+    );
+
+    const backgroundColorToken = await getDesignTokensByCssProperty(
+      page,
+      stepFlowProgressIndicator(page).nth(2),
+      "background-color"
+    );
+
+    expect(backgroundColorToken[0]).toBe("--colorsActionDisabled600");
+  });
+});
+
+test.describe("Event checks for StepFlow component", () => {
+  test("should call onDismiss callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <StepFlowComponent
+        showCloseIcon
+        onDismiss={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    await stepFlowDismissIcon(page).click();
+    expect(callbackCount).toBe(1);
+  });
+});
+
+test.describe("Ref checks for StepFlow component", () => {
+  test("should focus on title DOM element when a focus ref is passed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<StepFlowComponentWithRefAndButtons />);
+
+    await expect(stepFlowTitleTextWrapper(page)).not.toBeFocused();
+    await button(page).nth(1).click();
+    await expect(stepFlowTitleTextWrapper(page)).toBeFocused();
```

Comment:
nitpick(non-blocking): Same again with the broken legs.
```suggestion
    await button(page).nth(1).click();
    
    await expect(stepFlowTitleTextWrapper(page)).toBeFocused();
```